API/delivery/openapi.yaml

```diff
@@ -0,0 +1,283 @@
+      enum:
+        - choice
+        - email
+        - mood
+        - nps
+        - rating
+        - text
+        - polar
+      type: string
+    ElementIdentifier:
+      description: the unique text identifier for an element inside the form
+      type: string
+      maxLength: 200
+      minLength: 1
+      example: an_identifier_with_undescore_text
+    Element:
+      type: object
+      description: An element belonging to a form
+      properties:
+        id:
+          $ref: '#/components/schemas/Uuid'
+        type:
+          $ref: '#/components/schemas/ElementType'
+        identifier:
+          $ref: '#/components/schemas/ElementIdentifier'
+        title:
+          $ref: '#/components/schemas/Title'
+        required:
+          type: boolean
+          description: Whether a client displaying this element should force the user to fill it in before proceding
+      required:
+        - id
+        - type
+        - identifier
+    TextElement:
+      allOf:
+        - $ref: '#/components/schemas/Element'
+        - type: object
+          properties:
+            maxCharacterCount:
+              type: integer
+              minimum: -1
+              default: -1
+              maximum: 10000
+    RateableElement:
+      allOf:
+        - $ref: '#/components/schemas/Element'
+        - type: object
+          description: An element belonging to a form, that can contain multiple answer options
+          properties:
+            answerOptions:
+              type: array
+              items:
+                type: object
+                description: the answer options that rateable elements contains
+                properties:
+                  id:
+                    description: the UUID for the answer option
+                    allOf:
+                      - $ref: '#/components/schemas/Uuid'
+                  identifier:
+                    type: string
+                required:
+                  - id
+                  - identifier
+          required:
+            - answerOptions
+    ChoiceElement:
+      allOf:
+        - $ref: '#/components/schemas/RateableElement'
+        - type: object
+          description: An element belonging to a form, that can contain multiple answer options, and can allow for multiple answers to be selected by the end user
+          properties:
+            allowMultiple:
+              description: Indication whether this element allows multiple answer options to be selected
+              type: boolean
+          required:
+            - allowMultiple
+    NPSElement:
+      allOf:
+        - $ref: '#/components/schemas/RateableElement'
+        - type: object
+          description: An element that contains 11 answer options statically generated.
+          properties:
+            labelHigh:
+              $ref: '#/components/schemas/Label'
+            labelLow:
+              $ref: '#/components/schemas/Label'
+    SliderElement:
```

Comment:
Missing ranking

Reply:
I've added a different schema for each element just to make looking for a specific element schema easier.